keps/sig-auth/4601-authorize-with-selectors/README.md

```diff
@@ -0,0 +1,795 @@
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The authorization attributes will be extended to include field selectors and label selectors from
+Get, List, Watch, and Deletecollection.
+This will allow authorizers to use these selectors when making an authorization decision.
+
+## Motivation
+
+Security for per-node workloads could be improved by exposing field and label selectors to authorizers.
+Adding them as authorization attributes allows the development of new kinds of authorizers that
+leverage this information to provide security.
+In particular, it enables out-of-tree authorizers to experiment with ways to express restrictions based on field and label selectors.
+
+### Goals
+
+* Add field and label selectors to authorization attributes for Get, List, Watch, and DeleteCollection verbs.
+* Add field and label selectors to webhook authorization types.
+* Add field and label selectors to SelfSubjectAccessReview (SSAR) and SubjectAccessReview (SAR).
+  <<[UNRESOLVED deads2k, liggitt, aws-Micah: you guys volunteering to deliver this bit? ]>>
+* Update node authorizer to restrict on nodeName field selector.
+  <<[/UNRESOLVED]>>
+
+### Non-Goals
+
+* Create a generic in-tree authorizer that manages field or label selectors.
+
+## Proposal
+
+List, Watch, and DeleteCollection requests directly have field and label selector options.
+A single-item List or Watch request is remapped to the Get verb, so Get requests can have field and label selectors.
+
+### Authorization Attributes changes
+
+The authorization attributes have easy access to the query parameter field and label selectors.
+To avoid confusion, field and label selectors will not be included in authorization attributes for verbs where the field
+selector has no semantic meaning.
+In practice this means that (for now), only Get, List, Watch, and DeleteCollection have field and label selectors.
+Any authorizer that gets an error from `ParseFieldSelector` or `ParseLabelSelector` may attempt to authorize with the 
+`GetFieldSelector` or `GetLabelSelector` or attempt to authorize without field or label selectors since that will authorize
+using a wider permission (field and label selectors can only reduce access).
+
+```go
+type Attributes interface {
+	// GetFieldSelector returns the query parameter value of the field selector
+	// Remember that only List, Watch, and DeleteCollection (currently) provide 
+	// Remember that field selector formats vary based on the version of the API being used!
+	GetFieldSelector() string
+
+    // GetLabelSelector returns the query parameter value of the label selector
+	GetLabelSelector() string
```

Comment:
I'm not sure we need access to the raw strings in the interface now, if we authorized unrestricted when parse fails, and always sent only parsed requirements to webhooks